canopy_store.ml

```diff
@@ -110,21 +110,24 @@ module Store (C: CONSOLE) (CTX: Irmin_mirage.CONTEXT) (INFL: Git.Inflate.S) = st
     | Some a, Some b -> Lwt.return (a, b)
     | _ -> raise (Invalid_argument "date_updated_last")
 
-  let fill_cache article_hashtbl =
+  let fill_cache article_map =
     let open Canopy_content in
     let key_to_path key = List.fold_left (fun a b -> a ^ "/" ^ b) "" key in
     let fold_fn key value acc =
       value >>= fun content ->
       date_updated_created key >>= fun (updated, created) ->
       let uri = List.fold_left (fun s a -> s ^ "/" ^ a) "" key in
       match of_string ~uri ~content ~created ~updated with
-	| Ok article -> (KeyHashtbl.replace article_hashtbl key article; Lwt.return acc)
-	| Error error ->
-	   let error_msg = Printf.sprintf "Error while parsing %s: %s" (key_to_path key) error in
-	   Lwt.return (error_msg::acc)
-	| Unknown ->
-	   let error_msg = Printf.sprintf "%s : Unknown content type" (key_to_path key) in
-	   Lwt.return (error_msg::acc)
+      | Ok article -> (
+          article_map := KeyMap.remove key !article_map;
```

Comment:
`Canopy_dispatch` calls `store.update` which is exactly the `update` defined in `Canopy_main` :)